tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/service/BigQueryService.java

```diff
@@ -59,4 +76,29 @@ public BigQueryException catchExceptionFromDryRun(String query) {
 
     return null;
   }
+
+  /**
+   * Fetch the names of all table from the BigQuery given the project and dataset. If the project
+   * and/or dataset do not exist or not visible to the provided service account, {@link
+   * com.google.cloud.bigquery.BigQueryException} will be thrown.
+   *
+   * <p>It is possible that a dataset contains a large amount of tables, so this method will only
+   * fetch maximum 1000 of them.
```

Comment:
1000 seems a bit small.. :) Maybe you can adjust this later.

Reply:
Yes, I think I can make it as an optional argument later for users to specify.